app/src/main/java/org/wikipedia/settings/AppIconDialog.kt

```diff
@@ -0,0 +1,138 @@
+            flexDirection = FlexDirection.ROW
+            justifyContent = JustifyContent.CENTER
+            alignItems = AlignItems.CENTER
+        }
+        binding.appIconRecyclerView.apply {
+            this.layoutManager = layoutManager
+            adapter = appIconAdapter
+        }
+        appIconAdapter.updateItems(LauncherIcon.initialValues())
+    }
+
+    private fun updateIcons(selectedIcon: LauncherIcon) {
+        val currentSelectedIcon = if (Prefs.currentSelectedAppIcon != null) Prefs.currentSelectedAppIcon
+        else selectedIcon.key
+
+        LauncherIcon.entries.forEach {
+            it.isSelected = it.key == currentSelectedIcon
+        }
+        appIconAdapter.updateItems(LauncherIcon.entries)
+    }
+
+    private class AppIconAdapter : RecyclerView.Adapter<AppIconAdapter.AppIconViewHolder>() {
+        private var list = mutableListOf<LauncherIcon>()
+        private var onItemClickListener: ((LauncherIcon) -> Unit)? = null
+
+        fun onItemClickListener(onItemClickListener: (LauncherIcon) -> Unit) {
+            this.onItemClickListener = onItemClickListener
+        }
+
+        fun updateItems(newList: List<LauncherIcon>) {
+            list.clear()
+            list.addAll(newList)
+            notifyDataSetChanged()
+        }
+
+        override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): AppIconViewHolder {
+            val view = ItemAppIconBinding.inflate(LayoutInflater.from(parent.context), parent, false)
+            return AppIconViewHolder(view)
+        }
+
+        override fun getItemCount(): Int = list.size
+
+        override fun onBindViewHolder(holder: AppIconViewHolder, position: Int) {
+            val item = list[position]
+            holder.bind(item)
+        }
+
+        private inner class AppIconViewHolder(val binding: ItemAppIconBinding) : RecyclerView.ViewHolder(binding.root) {
+            fun bind(item: LauncherIcon) {
+                binding.appIcon.apply {
+                    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+                        foreground = ContextCompat.getDrawable(binding.root.context, item.foreground)
+                    } else {
+                        setImageDrawable(ContextCompat.getDrawable(binding.root.context, item.foreground))
+                    }
+                    background = ContextCompat.getDrawable(binding.root.context, item.background)
+                    setOnClickListener {
+                        onItemClickListener?.invoke(item)
+                    }
+                    val strokeColor = if (item.isSelected) {
+                        ContextCompat.getColor(binding.root.context, R.color.blue600)
+                    } else ContextCompat.getColor(binding.root.context, R.color.gray200)
+                    val newStrokeWidth = if (item.isSelected) 2f else 1f
+                    this.strokeColor = ColorStateList.valueOf(strokeColor)
+                    this.strokeWidth = newStrokeWidth
```

Comment:
The `strokeWidth` is not correct, please use `dpToPx` function for the correct value `DimenUtil.dpToPx(newStrokeWidth)`
and we should use theme-based style for the `strokeColor` too:

```
val strokeColor = if (item.isSelected) R.attr.progressive_color else R.attr.border_color
val newStrokeWidth = if (item.isSelected) 2f else 1f
this.strokeColor = ResourceUtil.getThemedColorStateList(context, strokeColor)
this.strokeWidth = DimenUtil.dpToPx(newStrokeWidth)
```

Reply:
done.